### PR TITLE
doc: Keepalive missing in sync worker and value recommendation

### PR DESCRIPTION
**Pull request by tuukkamustonen**
Related to: https://github.com/benoitc/gunicorn/issues/1194

As per https://github.com/benoitc/gunicorn/issues/1194#issuecomment-239917723 am I'm correct in stating that `sync` worker just doesn't support keep-alive, even if you try setting the header manually?

### CODE REVIEW

File: docs/source/settings.rst [hunk 1]
```diff
@@ -234,7 +234,13 @@ keepalive
 
 The number of seconds to wait for requests on a Keep-Alive connection.
 
-Generally set in the 1-5 seconds range.
+Generally set in the 1-5 seconds range for servers with direct connection
```
Comment (berkerpeksag) [inline docs/source/settings.rst]: Could you make this change in `gunicorn/config.py` instead and then run `make html` in `docs/` directory? (After running `make html` you will need to strip unrelated changes in `docs/source/settings.rst`.)

Comment (tuukkamustonen) [inline docs/source/settings.rst]: Ah, it gets copied from there... alright, thanks for the guidance... I adjusted PR accordingly.

**Comment by tilgovi** [comment]
Seems fine to me.

File: docs/source/design.rst [hunk 2]
```diff
@@ -34,6 +34,10 @@ as any errors will affect at most a single request. Though as we describe below
 only processing a single request at a time requires some assumptions about how
 applications are programmed.
 
+The sync worker does not support keep-alive - each connection is closed after
```
Comment (berkerpeksag) [inline docs/source/design.rst]: As a non-native speaker, I'd prefer "persistent connection" instead of keep-alive. On another hand, we already have a  `--keep-alive` setting so perhaps it's OK. @tilgovi, what do you think?

File: docs/source/design.rst [hunk 3]
```diff
@@ -34,6 +34,10 @@ as any errors will affect at most a single request. Though as we describe below
...
+response has been sent (even if you manually add ``Keep-Alive`` header in
```
Comment (berkerpeksag) [inline docs/source/design.rst]: Is `Keep-Alive` a standard now? We could probably mention `Connection: keep-alive` too.

File: docs/source/settings.rst [hunk 4]
```diff
@@ -234,7 +234,13 @@ keepalive
...
+to the client (=when you don't have separate load balancer). When
```
Comment (berkerpeksag) [inline docs/source/settings.rst]: Perhaps replace '=' with 'i.e.'?

File: docs/source/settings.rst [hunk 5]
```diff
@@ -234,7 +234,13 @@ keepalive
...
+gunicorn is deployed behind a load balancer, it often makes sense to
```
Comment (berkerpeksag) [inline docs/source/settings.rst]: gunicorn -> Gunicorn

File: docs/source/settings.rst [hunk 6]
```diff
@@ -234,7 +234,13 @@ keepalive
...
+set this to a higher value.
+
+Note that ``sync`` worker does not support keep-alive and will ignore
```
Comment (berkerpeksag) [inline docs/source/settings.rst]: We could use a note directive here:

```rst
.. note::
   ...
```

File: docs/source/design.rst (hunk 2 above)
Comment (tilgovi) [inline docs/source/design.rst]: I defer to you, since English is my first language. Your judgement is more helpful here.

File: docs/source/design.rst (hunk 3 above)
Comment (tilgovi) [inline docs/source/design.rst]: That's correct.

Comment (tilgovi) [inline docs/source/design.rst]: It is `Connection: keep-alive` as far as I know.

File: docs/source/settings.rst (hunk 4 above)
Comment (tilgovi) [inline docs/source/settings.rst]: Or 'e.g.', since it might be something other than a load balancer (like an app server that has a connection to another backend service that uses gunicorn).

**Comment by tilgovi** [comment]
Great comments, @berkerpeksag!

**Comment by tuukkamustonen** [comment]
@tilgovi @berkerpeksag Sorry for the quite drastic delay... the PR has been updated accordingly.

File: docs/source/design.rst [hunk 7]
```diff
@@ -34,6 +34,10 @@ as any errors will affect at most a single request. Though as we describe below
 only processing a single request at a time requires some assumptions about how
 applications are programmed.
 
+The sync worker does not support peristent connections - each connection is
```
Comment (berkerpeksag) [inline docs/source/design.rst]: peristent -> persistent

File: gunicorn/config.py [hunk 8]
```diff
@@ -760,7 +760,14 @@ class Keepalive(Setting):
...
         The number of seconds to wait for requests on a Keep-Alive connection.
 
-        Generally set in the 1-5 seconds range.
+        Generally set in the 1-5 seconds range for servers with direct connection
+        to the client (e.g. when you don't have separate load balancer). When
+        Gunicorn is deployed behind a load balancer, it often makes sense to
+        set this to a higher value.
+
+        .. note::
+           ``sync`` worker does not support keep-alive and will ignore this
```
Comment (berkerpeksag) [inline gunicorn/config.py]: We could use the same wording in design.rst and say 

```rst
``sync`` worker does not support persistent connections [...]
```

**Comment by tuukkamustonen** [comment]
Oopsie. Maybe now?

**Comment by berkerpeksag** [comment]
Looks great, thank you for improving the documentation! :)
